gateway/s3x/gateway-s3x-object.go

```diff
@@ -11,43 +11,53 @@ import (
 	minio "github.com/minio/minio/cmd"
 )
 
-// ListObjects lists all blobs in S3 bucket filtered by prefix
+// ListObjects lists all objects in a bucket
+// The prefix and delimiter parameters limit the kind of results returned by a list operation.
+// The prefix limits the results to only those keys that begin with the specified prefix.
+// The delimiter causes a list operation to roll up all the keys that share a common prefix into a single summary list result.
 func (x *xObjects) ListObjects(
 	ctx context.Context,
 	bucket, prefix, marker, delimiter string,
 	maxKeys int,
 ) (loi minio.ListObjectsInfo, e error) {
-	// TODO(bonedaddy): implement complex search (George: prefix implemented)
-	objs, err := x.ledgerStore.GetObjectInfos(ctx, bucket, prefix, "", 0)
+	objs, folders, nextMarker, err := x.ledgerStore.GetObjectInfos(ctx, bucket, prefix, marker, delimiter, maxKeys)
 	if err != nil {
 		return loi, x.toMinioErr(err, bucket, "", "")
 	}
 	loi.Objects = make([]minio.ObjectInfo, 0, len(objs))
 	for _, obj := range objs {
 		loi.Objects = append(loi.Objects, getMinioObjectInfo(&obj))
 	}
```

Comment:
I wonder, should we do:

```

	loi.Objects = make([]minio.ObjectInfo, len(objs))
	for i, obj := range objs {
		loi.Objects[i] = getMinioObjectInfo(&obj)
	}
```

In theory it would be slightly more efficient, since `append` will have to make sure the backing array has enough capacity?

Reply:
good idea